microprofile/connectors/kafka/src/main/java/io/helidon/microprofile/connectors/kafka/KafkaConfigProperties.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.connectors.kafka;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import io.helidon.config.Config;
+
+/**
+ * Prepare Kafka properties from Helidon {@link io.helidon.config.Config Config}.
+ * Configuration format as specified in the MicroProfile Reactive Messaging
+ * Specification https://github.com/eclipse/microprofile-reactive-messaging
+ *
+ * <p>
+ * See example with YAML configuration:
+ * <pre>{@code
+ * mp.messaging:
+ *   incoming:
+ *     test-channel:
+ *       bootstrap.servers: localhost:9092
+ *       topic: graph-done
+ *       key.deserializer: org.apache.kafka.common.serialization.LongDeserializer
+ *       value.deserializer: org.apache.kafka.common.serialization.StringDeserializer
+ *
+ *   outgoing:
+ *     test-channel:
+ *       bootstrap.servers: localhost:9092
+ *       topic: graph-done
+ *       key.serializer: org.apache.kafka.common.serialization.LongSerializer
+ *       value.serializer: org.apache.kafka.common.serialization.StringSerializer
+ *
+ * }</pre>
+ * <p>
+ *
+ * @see io.helidon.config.Config
+ */
+class KafkaConfigProperties extends Properties {
+
+    /**
+     * Topic or topics delimited by commas.
+     */
+    static final String TOPIC_NAME = "topic";
+
+    /**
+     * Consumer group id.
+     */
+    static final String GROUP_ID = "group.id";
+
+    /**
+     * Prepare Kafka properties from Helidon {@link io.helidon.config.Config Config},
+     * underscores in keys are translated to dots.
+     *
+     * @param config parent config of kafka key
+     */
+    KafkaConfigProperties(Config config) {
+        config.asNodeList().get().forEach(this::addProperty);
+    }
+
+    /**
+     * Split comma separated topic names.
+     *
+     * @return list of topic names
+     */
+    public List<String> getTopicNameList() {
+        return Arrays.stream(getProperty(TOPIC_NAME)
+                .split(","))
+                .map(String::trim)
+                .collect(Collectors.toList());
+    }
+
+    private void addProperty(Config c) {
```

Comment:
This is not the correct way to do this (I mentioned this in my previous review.

Reply:
Sorry I didn't notice github was hiding some comments.